mule-extensions-api-persistence/src/main/java/org/mule/runtime/extension/internal/persistence/ModelPropertyMapTypeAdapter.java

```diff
@@ -36,6 +38,17 @@
  */
 public final class ModelPropertyMapTypeAdapter extends TypeAdapter<HierarchyClassMap<ModelProperty>> {
 
+  private static final Map<Class<? extends ModelProperty>, String> classNameMapping;
+  private static final Map<String, Class<? extends ModelProperty>> nameClassMapping;
+
+  static {
+    classNameMapping = new HashMap<>();
```

Comment:
Missing mapping for `MetadataKeyPartModelProperty` and `MetadataKeyIdModelProperty`